packages/qwik/src/optimizer/src/plugins/plugin.ts

```diff
@@ -875,8 +875,38 @@ export const manifest = ${JSON.stringify(manifest)};\n`;
   // order by discovery time, so that related segments are more likely to group together
   function manualChunks(id: string, { getModuleInfo }: Rollup.ManualChunkMeta) {
     const module = getModuleInfo(id)!;
-    const segment = module.meta.segment as SegmentAnalysis | undefined;
-    return segment?.entry;
+    const segment = module.meta.segment;
+
+    const optimizer = getOptimizer();
+    const path = optimizer.sys.path;
+    const relativePath = path.relative(optimizer.sys.cwd(), id);
+    const sanitizedPath = relativePath.replace(/^\/+/, '').replace(/\//g, '-');
+
+    if (sanitizedPath.includes('node_modules')) {
```

Comment:
actually I wonder if we shouldn't bundle per node_module name - all the qrls will have been separately done already.

Reply:
That would be very problematic for big libraries that are properly bundled by rollup. For example with shiki, rollup currently creates one chunk per language syntax (e.g. js, ts, python, c, etc.), but if we tell it to bundle all the shiki stuff together rollup will produce one giant 10mb chunk.